lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1574,12 +1589,19 @@ void IRGenDebugInfoImpl::setCurrentLoc(IRBuilder &Builder,
     // Reuse the last source location if we are still in the same
     // scope to get a more contiguous line table.
     L = LastDebugLoc;
+  } else if (DS == LastScope && Loc.is<ArtificialUnreachableLocation>() &&
+             Opts.DebugInfoFormat == IRGenDebugInfoFormat::CodeView) {
+    // Remove artificial unreachable locations to make CodeView linetables
+    // more contiguous.
```

Comment:
Please also explain *why* this is important and the right trade-off for this platform.